JAVASCRIPT_FUNDAMENTALS/_Functions/js/main.js

```diff
@@ -0,0 +1,305 @@
+// Function declaration
+
+function showMessage() {
+    alert('Hello Everyone!')
+}
+
+showMessage(); //calling the function
+showMessage();
+
+
+
+// The function keyword goes first, then the name of the function, then the list of the parameters and finally
+//  the function body inside the curly braces
+
+// Functions avoid code duplication
+
+
+
+// Local Variables - a variable declared inside a function is only visible within that function
+
+function showMessage() {
+    let message = 'Hello Javascript';
+    alert(message);
+}
+
+showMessage(); //Hello, I'm Javascript
+alert(message); //Error! The variable is local to the function
+
+
+
+
+// outer variables
+
+// A function can access an outer variable as well
+
+
+let userName  = 'Annette';
+
+function showMessage() {
+    let message = 'Hello' + userName;
+    alert(message);
+}
+
+showMessage(); //Hello Annette
+
+
+// A function can modify an outer variable as well
+
+let userName = 'John';
+
+function showMessage(){
+    userName = Bob;
+    let message = 'Hello' + userName;
+    alert(message);
+}
+
+alert(userName); //John; before function call
+
+showMessage();
+alert(userName); //Bob; the function will modify the name
+
+
+// The outer variable is used if there is no outer one
+
+// If same-named variable is declared inside a function, then it overshadows the outer one e.g
+// the local variable is used below and the outer one is ignored
+
+let userName = 'Johnny';
+
+function showMessage() {
+    let userName = 'Annette'
+    let message = `Hello ${userName}`
+
+    alert(message);
+}
+
+showMessage(); //the function will create and use its own variable
+
+alert(userName); //Annette; the function did not read the outer variable
```

Comment:
@Annettesunday I ran this on https://jsfiddle.net/ and got a different result. Kindly confirm.

Reply:
@RufusMbugua It is different because the alert was being called outside of the function scope hence it was reading the global userName. I have fixed this though.